### PR TITLE
[Merged by Bors] - chore(RingTheory/JacobsonIdeal): split out Polynomial

### PR DESCRIPTION
**Pull request by alreadydone**
Also create a folder RingTheory/Jacobson containing three files Ideal (originally JacobsonIdeal), Ring (original Jacobson), and Polynomial (new).

22 files have transitive imports reduced by 100 or more.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alreadydone** [comment]
!bench

**Comment by alreadydone** [comment]
!bench

**Comment by alreadydone** [comment]
!bench

**Comment by alreadydone** [comment]
!bench

File: Mathlib/RingTheory/Ideal/Over.lean [hunk 1]
```diff
@@ -88,7 +88,7 @@ theorem injective_quotient_le_comap_map (P : Ideal R[X]) :
 R[x] / P → (R / (P ∩ R))[x] / (P / (P ∩ R))
 ```
 commutes.  It is used, for instance, in the proof of `quotient_mk_comp_C_is_integral_of_jacobson`,
-in the file `RingTheory.Jacobson`.
+in the file `RingTheory.Jacobson.Polynomial`.
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/Ideal/Over.lean:91]:
```suggestion
in the file `Mathlib.RingTheory.Jacobson.Polynomial`.
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Looks great!

**Comment by Ruben-VandeVelde** [comment]
maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**